### PR TITLE
1260 dfa add csv result files

### PR DESCRIPTION
**Pull request by fm-117**

### CODE REVIEW

**Review by smedilol — APPROVED** [review]

**Review by mayanje — APPROVED** [review]

**Review by rooksdo — CHANGES_REQUESTED** [review]
It think it would be interesting to get DFA results from the following test targetting specifically data description reorganization :
```cob
       IDENTIFICATION DIVISION.
       PROGRAM-ID. test.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 k0.
         02 i0    pic 9.
         02 j0    pic 9.
      
       01 NumericGroup1.
         02 i1    pic 9.
         02 j1    pic 9.
         66 k1 renames i1 thru j1.
      
       01 NumericGroup2.
         02 i2    pic 9.
         02 j2    pic 9.
       01 k2 redefines NumericGroup2 pic 99.
      
       01 k3.
         02 ktab occurs 4 indexed by i3 pic 9.
      
       PROCEDURE DIVISION.
      
      *   Testing group
           Move 1       to i0
           Move i0      to j0
           Add  1       to j1
      *    is j0 affected ?
           Move 0       to k0
           Move 11      to k0
      
      *   Testing renames
           Move 1       to i1
           Move i1      to j1
           Add  1       to j1
           Move 0       to k1
           Move 11      to k1
      
      *   Testing redefines
           Move 1       to i2
           Move i1      to j2
           Add  1       to j2
           Move 0       to k2
           Move 11      to k2
           Perform varying k2 from 1 by 1 until
                                   i2 > 9
              display 'k2 = ' i2 j2
           End-Perform
      
      *   Testing occurs
           Move 1111    to k3
           Perform varying i3 from 1 by 1 until
                                   i3 > 4
              Add  1    to ktab (i3)
           End-Perform
           .
       END PROGRAM test.
```

File: TypeCobol.Analysis.Test/BasicDfaSamples/SampleGotos0.dfa.csv [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Block2;1;    IF J = 999 THEN;;
+Block3;0;         GO TO L4;;
+Block6;0;    ADD 1 TO J;;
+Block6;1;    IF J = 999 THEN;;
+Block7;0;         GO TO L4;;
+Block9;0;    SUBTRACT 4 FROM J;;
+Block10;0;    GO TO L2;;
+Block;In;Out;Gen;Kill
+Block1;{};{I:Block1:0, J:Block1:2};{I:Block1:0, J:Block1:2};{I:Block2:0, J:Block6:0, J:Block9:0}
+Block2;{I:Block1:0, J:Block1:2, I:Block2:0, J:Block6:0, J:Block9:0};{J:Block1:2, I:Block2:0, J:Block6:0, J:Block9:0};{I:Block2:0};{I:Block1:0}
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol.Analysis.Test/BasicDfaSamples/SampleGotos0.dfa.csv:16]: I was confused about that too but @mayanje explained it that way: the Kill set is the set of all 'potential' redefinitions so it does not really take into account actual edges but instead it aggregates all the definitions of the target variable outside of the block.

[the remark this answers is not in the document]
Comment (rooksdo) [reply TypeCobol.Analysis.Test/BasicDfaSamples/SampleGotos0.dfa.csv:16]: So it is compliant with DFA specification, OK.

**Comment by fm-117** [comment]
@rooksdo, about the test you suggested. I think it is too advanced as for current DFA implementation. Here are the results:
[DfaAdvancedSample.cbl.txt](https://github.com/TypeCobolTeam/TypeCobol/files/5652763/DfaAdvancedSample.cbl.txt)
[DfaAdvancedSample.dot.txt](https://github.com/TypeCobolTeam/TypeCobol/files/5652761/DfaAdvancedSample.dot.txt)
[DfaAdvancedSample.xlsx](https://github.com/TypeCobolTeam/TypeCobol/files/5652762/DfaAdvancedSample.xlsx)
Some remarks:
- Renames are not supported so I commented them out
- Parent variables are not supported (i.e. modifying a child variable does not register as modifying the whole group, see #1807)
- UseList seems incomplete: `k2` in `perform varying k2` and `i3` in `perform varying i3` are not shown in the use-list, same goes for `ktab`.

**Review by rooksdo — APPROVED** [review]

**Pull request merged**